assets/js/blocks/product-query/constants.ts

```diff
@@ -73,6 +73,7 @@ export const QUERY_DEFAULT_ATTRIBUTES: QueryBlockAttributes = {
 };
 
 export const INNER_BLOCKS_TEMPLATE: InnerBlockTemplate[] = [
+	[ 'woocommerce/catalog-sorting' ],
```

Comment:
With the block being the wrapper for the PHP template function, it shouldn't be the inner block of the Products block. Also, catalog sorting only works in archive templates only.